src/api/app/views/webui/user/user_profile_redesign/_involvement.html.haml

```diff
@@ -50,6 +50,12 @@
           %button.btn.btn-primary.form-control#filter-button{ type: :submit }
             Apply Filter
 
+    - if display_filters?(filters.keys)
```

Comment:
Having another look, I'd suggest moving all this code to the helper.
In the view we can simply call one helper method, for example: `= clear_filters_link(filters.keys)`.
And inside the helper method do all the checks and return a `span` containing the link or just a link.

Reply:
hi @saraycp, i did move the code to a helper as suggested, i just let the check if we have to display the filters or not in the partial to make easier to understand the logic while reading the partial

Reply:
We can get rid of that helper method and use the following condition in the view instead:

`- unless filters.keys == ['search_text']`

AFAIK filter is never empty. If we don't apply any filter or search text, the `filters` variable contains `{ "search_text"=>nil }`.

So that condition will be true if there are no filters applied or if the only filter is some text, `{ "search_text"=>"some text" }`.

In the rest of the cases, the `filters` variable will have more than one key, so you can display the link.

Reply:
let me check

Reply:
right, could you please check it one more time @saraycp ?

Reply:
Ok. I would have moved it to the view, but I can understand that the method name is more meaningful than the condition itself. :+1: